IPython/core/interactiveshell.py

```diff
@@ -2068,13 +2068,17 @@ def showtraceback(self, exc_tuple=None, filename=None, tb_offset=None,
         except KeyboardInterrupt:
             print('\n' + self.get_exception_only(), file=sys.stderr)
 
-    def _showtraceback(self, etype, evalue, stb):
+    def _showtraceback(self, etype, evalue, stb: str):
         """Actually show a traceback.
 
         Subclasses may override this method to put the traceback on a different
         place, like a side channel.
         """
-        print(self.InteractiveTB.stb2text(stb))
+        val = self.InteractiveTB.stb2text(stb)
+        try:
+            print(val)
+        except UnicodeEncodeError:
+            print(val.encode("utf-8", "backslashreplace").decode())
```

Comment:
What are the benefits of having `try...except` here rather than just `print(val.encode(...).decode()`?

Reply:
I believe this is question of efficiency. In most case `print()` will work, so there is no need to do the expensive `encode()`, then `.decode()` dance, that would in the end often be no-op.